nemo/collections/asr/models/rnnt_spk_bpe_models.py

```diff
@@ -115,14 +177,89 @@ def _setup_transcribe_dataloader(self, config: Dict) -> 'torch.utils.data.DataLo
             'num_speakers': self.cfg.test_ds.get('num_speakers',4),
             'spk_tar_all_zero': self.cfg.test_ds.get('spk_tar_all_zero',False),
             'num_sample_per_mel_frame': self.cfg.test_ds.get('num_sample_per_mel_frame',160),
-            'num_mel_frame_per_asr_frame': self.cfg.test_ds.get('num_mel_frame_per_asr_frame',8)
+            'num_mel_frame_per_asr_frame': self.cfg.test_ds.get('num_mel_frame_per_asr_frame',8),
+            'shuffle_spk_mapping': self.cfg.test_ds.get('shuffle_spk_mapping',False)
         }
 
         if config.get("augmentor"):
             dl_config['augmentor'] = config.get("augmentor")
 
         temporary_datalayer = self._setup_dataloader_from_config(config=DictConfig(dl_config))
         return temporary_datalayer
+    
+    def forward_diar(
+        self,
+        input_signal=None,
+        input_signal_length=None,
+    ):
+        preds, _preds, attn_score_stack, total_memory_list, encoder_states_list = self.diarization_model.forward(audio_signal=input_signal, audio_signal_length=input_signal_length)
+
+        return preds, _preds, attn_score_stack, total_memory_list, encoder_states_list
+
+    def fix_diar_output(
```

Comment:
Do we still need this function as you implement that get_hidden_length function to get the actual target length?

Reply:
I think this is meant to resolve difference between asr encoder vs diar_preds difference. diar_preds are not from get_hidden_length_function, but from sortformer.